src/components/ChallengeModal/ChallengeModal.utils.ts

```diff
@@ -207,7 +207,7 @@ export function rejectionDetailsToMessage(details: RejectionDetails): string | u
 
 // For legacy reasons, handicap can have a string value
 // when these challenge details come from local storage.
-export function sanitizeChallengeDetails(challengeDetails: ChallengeDetails): ChallengeDetails {
+export function sanitizeChallengeDetails<T extends ChallengeDetails>(challengeDetails: T): T {
```

Comment:
This change occured to me during the process. It is not necessary here, but I believe it's worth keeping.